php/WP_CLI/Runner.php

```diff
@@ -326,8 +325,18 @@ public function run_command( $args, $assoc_args = array(), $options = array() )
 		}
 	}
 
-	private function _run_command() {
+	private function _run_command_and_exit() {
 		$this->run_command( $this->arguments, $this->assoc_args );
+		if ( $this->cmd_starts_with( array( 'help' ) ) ) {
+			// Help couldn't find the command so exit with suggestion.
+			$possible_suggestion = $this->find_command_to_run( array_slice( $this->arguments, 1 ) );
+			if ( is_string( $possible_suggestion ) ) {
+				WP_CLI::error( $possible_suggestion );
+			}
+			// Should never happen.
```

Comment:
Removed amazingly amusing error message and now `Help_Command` uses `Runner::find_command_to_run()` so should be 100% confident.